enhancements/pipeline-tooling.md

```diff
@@ -0,0 +1,267 @@
+    │   │   └── bundle.clusterserviceversion.yaml
+    │   ├── v2
+    │   │   ├── some.crd.yaml
+    │   │   ├── some.other.crd.yaml
+    │   │   ├── bundle.v2.clusterserviceversion.yaml
+    │   └── bundle.package.yaml
+    ```
+
+* takes a manifest directory for a given operator and migrates it to an individual bundle
+
+    Part of this is already implemented in the `opm alpha bundle build` command which takes the manifest for a specific version - for example `path/to/my/dir/bundle/v1` from above:
+    
+    ```
+    dir
+    ├── bundle
+    │   ├── v1
+    │   │   ├── some.crd.yaml
+    │   │   └── bundle.clusterserviceversion.yaml
+    ```
+    
+    and converts that to the new bundle format - including a Dockerfile to push this bundle to a container registry.
+    
+    Q: does it make sense to implement a wrapper for this command that takes a manifest directory instead of a single version to output an index or multiple bundles?
+
+### Download the contents of an index
+
+As an Operator Registry maintainer, Pipeline admin, or QE, I want to download or inspect the contents of a given index image.
+
+#### Design Proposal
+
+Add a command to `opm index` and `opm alpha bundle` called `extract` which extracts:
+
+* the manifest and metadata folders for a given bundle
+
+    **Example CLI:**
+    
+    `opm alpha bundle extract quay.io/my/bundle:v1 --out path/to/my/dir`
+
+* a list of all the bundle images that are in an index
+    
+    **Example CLI 1:**
+    
+    `opm index extract quay.io/my/index --out path/to/my/dir --images-only`
+    
+    **Example CLI 2:** prints to stdout
+    
+    `opm alpha bundle extract quay.io/my/bundle:v1 --image-only`
+
+
+* the manifest and metadata folders for all bundles in a given index
+
+    **Example CLI:**
+    
+    `opm index extract quay.io/my/index --out path/to/my/dir`
+
+
+### Delete latest operator version
+
+As an Operator Registry maintainer, I want to have a way to remove the latest version of an operator (from an index)?
+
+#### Design Proposal
+
+Add a new flag to `opm registry rm` and `opm index rm` commands called `--latest` which:
+
+* takes an index reference and package name, and removes the latest bundle version if present.
+    
+    **Example CLI:**
+    `opm index rm --latest --from quay.io/my/index -o "operator1,operator2" -c=docker`
+
+* takes a registry database and package names, and removes the latest bundle version if present.
+
+    **Example CLI:**
+    `opm registry rm --latest -d=path/to/my.db -o="operator1,operator2"`
+    
+### Merging Indexes
+
+As an Operator Registry Maintainer I want to merge two indexes into one index.
+
+#### Design Proposal
+
+Initially we will only support disjoint indexes (i.e. indexes with no common packages) so that we don't have to deal with compatible versions and insertion order (which we currently cannot do implicitly). Once we change our update graph to rely solely on semver we could support merging intersecting indexes.
```

Comment:
Even if we had the versioning, is merging intersecting indexes valuable? Do we have an actual usecase for that pattern?